nextcord/guild.py

```diff
@@ -2902,7 +2913,25 @@ async def ban(
         HTTPException
             Banning failed.
         """
-        await self._state.http.ban(user.id, self.id, delete_message_days, reason=reason)
+        if delete_message_days is not None and delete_message_seconds is not None:
+            raise InvalidArgument(
+                "Cannot pass both delete_message_days and delete_message_seconds."
+            )
+        elif delete_message_days is None and delete_message_seconds is None:
+            # Default to one day
+            delete_message_seconds = 24 * 60 * 60
+
+        if delete_message_days is not None:
+            warnings.warn(
+                DeprecationWarning(
+                    "delete_message_days is deprecated, use delete_message_seconds instead."
+                )
+            )
+            delete_message_seconds = delete_message_days * 24 * 60 * 60
+
+        await self._state.http.ban(
+            user.id, self.id, cast(int, delete_message_seconds), reason=reason
+        )
```

Comment:
You don't need to use `cast` if you have a condition that checks if `delete_message_seconds` is `None`.

```suggestion
        elif delete_message_days is not None:
            warnings.warn(
                DeprecationWarning(
                    "delete_message_days is deprecated, use delete_message_seconds instead."
                )
            )
            delete_message_seconds = delete_message_days * 24 * 60 * 60
        elif delete_message_seconds is None:
            # Default to one day
            delete_message_seconds = 24 * 60 * 60

        await self._state.http.ban(user.id, self.id, delete_message_seconds, reason=reason)
```